syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/src/org/fusesource/ide/syndesis/extensions/ui/wizards/pages/SyndesisExtensionProjectWizardExtensionDetailsPage.java

```diff
@@ -0,0 +1,216 @@
+		
+		Label extensionIconLabel = new Label(container, SWT.NONE);
+		extensionIconLabel.setText(Messages.newProjectWizardExtensionDetailsPageIconLabel);
+		gridData = new GridData(SWT.FILL, SWT.TOP, true, false, 1, 1);
+		extensionIconText = new Text(container, SWT.BORDER);
+		extensionIconText.setLayoutData(gridData);
+		extensionIconText.setToolTipText(Messages.newProjectWizardExtensionDetailsPageIconTooltip);
+		extensionIconText.addFocusListener(focusListener);
+		Button extensionIconBrowseButton = new Button(container, SWT.PUSH | SWT.BORDER);
+		extensionIconBrowseButton.setText(Messages.newProjectWizardExtensionDetailsPageIconBrowseLabel);
+		extensionIconBrowseButton.addFocusListener(focusListener);
+		extensionIconBrowseButton.addSelectionListener( new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent e) {
+				FileDialog dialog = new FileDialog(getShell());
+				dialog.setText(Messages.newProjectWizardExtensionDetailsPageIconSelectionDialogTitle);
+				dialog.setFilterPath(".");
+				dialog.setFilterExtensions(new String[] { "*.gif;*.png;*.jpg;*.jpeg;*.svg" });
+				dialog.setFilterNames(new String[] { Messages.newProjectWizardExtensionDetailsPageIconSelectionDialogFileTypeLabel });
+				String selectedIcon = dialog.open();
+				if (selectedIcon != null) {
+					File icon = new File(selectedIcon);
+					if (icon.exists() && icon.isFile() && icon.canRead()) {
+						extensionIconText.setText(selectedIcon);						
+					} else {
+						setErrorMessage(NLS.bind(Messages.newProjectWizardExtensionDetailsPageIconSelectionDialogFileUnavailableError, selectedIcon));
+					}
+				}
+			}
+		});
+		
+		Label extensionTagsLabel = new Label(container, SWT.NONE);
+		extensionTagsLabel.setText(Messages.newProjectWizardExtensionDetailsPageTagsLabel);
+		gridData = new GridData(SWT.FILL, SWT.TOP, true, false, 2, 1);
+		extensionTagsText = new Text(container, SWT.BORDER);
+		extensionTagsText.setLayoutData(gridData);
+		extensionTagsText.setToolTipText(Messages.newProjectWizardExtensionDetailsPageTagsTooltip);
+		extensionTagsText.addFocusListener(focusListener);
+		
+		setControl(container);
+		
+		extensionIdText.setFocus();
+	}
+
+	public String getExtensionId() {
+		if (!Widgets.isDisposed(extensionIdText) && !Strings.isBlank(extensionIdText.getText())) {
+			return extensionIdText.getText();
+		}
+		return null;
+	}
+	
+	public String getExtensionVersion() {
+		if (!Widgets.isDisposed(extensionVersionText) && !Strings.isBlank(extensionVersionText.getText())) {
+			return extensionVersionText.getText();
+		}
+		return null;
+	}
+	
+	public String getExtensionName() {
+		if (!Widgets.isDisposed(extensionNameText) && !Strings.isBlank(extensionNameText.getText())) {
+			return extensionNameText.getText();
+		}
+		return null;
+	}
+	
+	public String getExtensionDescription() {
+		if (!Widgets.isDisposed(extensionDescriptionText) && !Strings.isBlank(extensionDescriptionText.getText())) {
```

Comment:
I don't see an issue here tbh. If it returns null because of an unlikely disposed event then the extension will contain null as description which leads to an empty description.

Reply:
I have more in mind trying to use this value after the wizard is closed so would mean that it will be always disposed. But even in your use case it means that the value displayed is wrong, it would be an empty description instead of the correct description.